arrows/vxl/hashed_image_classifier.cxx

```diff
@@ -0,0 +1,498 @@
+      {
+        weight_t& output{ output_image( i, j ) };
+
+        output = offset;
+
+        for( unsigned f = 0; f < features; ++f )
+        {
+          output += feature_weights[ f ][ input_features[ f ]( i, j ) ];
+        }
+      }
+    }
+  }
+}
+
+// ----------------------------------------------------------------------------
+// Load a model from file
+template < typename FeatureType, typename OutputType >
+bool
+hashed_image_classifier< FeatureType, OutputType >
+::load_from_file( std::string const& file )
+{
+  model_.reset( new model_t{} );
+
+  // Open model file for reading
+  std::ifstream input( file.c_str() );
+
+  if( !input.is_open() )
+  {
+    LOG_ERROR( logger, "Unable to open input file: " << file );
+    return false;
+  }
+
+  // Create a new vector to initially read our loaded model into
+  std::vector< std::vector< weight_t > > weights;
+
+  // Read each feature entry
+  unsigned entry = 0;
+  bool size_read_ = false;
+
+  std::string line;
+  while( std::getline( input, line ) )
+  {
+    // Parse line
+    std::vector< std::string > parsed;
+    std::istringstream iss{ line };
+
+    while( iss )
+    {
+      std::string word;
+      iss >> word;
+      parsed.push_back( word );
+    }
+
+    // Make sure last entry is not an empty char
+    if( parsed.size() > 0 )
+    {
+      std::string last = parsed[ parsed.size() - 1 ];
+      if( last == " " || last == "" || last == "\n" )
+      {
+        parsed.pop_back();
+      }
+    }
+
+    // Skip empty lines
+    if( parsed.size() == 0 )
+    {
+      continue;
+    }
+
+    // Skip comments indicated by pound symbol
+    if( parsed[ 0 ].size() > 0 && parsed[ 0 ][ 0 ] == '#' )
+    {
+      continue;
+    }
+
+    // Read feature count first
+    if( !size_read_ )
+    {
+      // Read number of features
+      model_->num_features =
+        static_cast< unsigned int >( std::stoi( parsed[ 0 ] ) );
+
+      // Make sure the model file has at least 1 input feature
+      if( model_->num_features == 0 )
+      {
+        LOG_ERROR( logger,
+                   "Number of input features to use must be > 1 but was " <<
+                   model_->num_features );
+        return false;
+      }
+
+      weights.resize( model_->num_features, std::vector< weight_t >{} );
+
+      size_read_ = true;
+      continue;
+    }
+
+    // Validate row size
+    if( parsed.size() < 2 )
+    {
+      continue;
+    }
+
+    // Read number of possible values for this feature
+    unsigned int num_values =
+      static_cast< unsigned int >( std::stoi( parsed[ 0 ] ) );
+    weights[ entry ].resize( num_values, 0.0 );
+
+    // Data corruption, model file ill formatted
+    if( parsed.size() != num_values + 1  || entry >= model_->num_features )
+    {
+      LOG_ERROR( logger,
+                 "Number of weights ("  << parsed.size() - 1
+                                        << ") does not match "
+                                        << num_values );
```

Comment:
```suggestion
                 "Number of weights (" << parsed.size() - 1
                                       << ") does not match "
                                       << num_values );
```
(Is this uncrustify being wonky, or did this just get missed?)

Reply:
Yes, uncrustify breaks this. I think I've seen two spaces between string litterals and `<<` in several places where there are multi-line outputs. Interestingly, if you try to pack these lines as tightly as possible and run `uncrustify`, you get the following: 
```
      LOG_ERROR( logger,
                 "Number of weights ("  << parsed.size() - 1
                                        << ") does not match " <<
            num_values );
```